src/server/ftpserver/listen.rs

```diff
@@ -35,17 +39,41 @@ where
             options,
             shutdown_topic,
             failed_logins,
+            connection_helper,
+            connection_helper_args,
         } = self;
         let listener = TcpListener::bind(bind_address).await?;
         loop {
             let shutdown_listener = shutdown_topic.subscribe().await;
             match listener.accept().await {
                 Ok((tcp_stream, socket_addr)) => {
                     slog::info!(logger, "Incoming control connection from {:?}", socket_addr);
-                    let result =
-                        controlchan::spawn_loop::<Storage, User>((&options).into(), tcp_stream, None, None, shutdown_listener, failed_logins.clone()).await;
-                    if let Err(err) = result {
-                        slog::error!(logger, "Could not spawn control channel loop for connection from {:?}: {:?}", socket_addr, err)
+                    if let Some(helper) = connection_helper.as_ref() {
+                        slog::info!(logger, "Spawning {:?}", helper);
+                        let fd = tcp_stream.as_raw_fd();
+                        nix::fcntl::fcntl(fd, nix::fcntl::FcntlArg::F_SETFD(nix::fcntl::FdFlag::empty())).unwrap();
+                        let result = tokio::process::Command::new(helper)
+                            .args(connection_helper_args.iter())
```

Comment:
OK from you commit message is seems to be about `kqueue`:

>     The server must have an optional connection_helper program that it will exec
      to handle each connected socket.  This is necessary so that users with
      different home directories can be handled by processes that are jailed
      to their specific home directories.  It's necessary to exec, rather
      than merely fork, because Tokio uses kqueue, and a kqueue is not
      inherited by children.

Reply:
Yes, that's right.  And the helper program does indeed invoke libunftp.  See crates/unftp-sbe-fs/examples/cap-ftpd-worker.rs .  It calls `Server::service`.